### ISSUE TITLE
x/pkgsite: hyperlink does not inherit version for standard library packages.

### ISSUE THREAD

**Issue by alexbozhenko** [golang/go#68755]
### What is the URL of the page with the issue?

https://pkg.go.dev/iter@go1.23rc2#hdr-Standard_Library_Usage

### What is your user agent?

Mozilla/5.0 (X11; Linux x86_64; rv:128.0) Gecko/20100101 Firefox/128.0

### Screenshot

Working link to the same pakcage, includes version:
![image](https://github.com/user-attachments/assets/fb370cbd-bae9-47fa-9f99-3443f5917c7f)

Broken link to a different package, does not include version:
![image](https://github.com/user-attachments/assets/0fafbd20-1fa2-4f6b-a8cd-72ddb4c56784)

### What did you do?

1. Open https://pkg.go.dev/iter@go1.23rc2#hdr-Standard_Library_Usage
2. Click on the link in  `For example, [maps.Keys](https://pkg.go.dev/maps#Keys) returns an iterator over the keys of a map`

### What did you see happen?

Observe that link points to the latest(go1.22.6), which does not contain `maps.Keys` function. Anchor link is broken. 
Note 1.23rc2 does contain that function.

### What did you expect to see?

I expected that if I explicitly chose to browse version 1.23rc2, all links would be pointing to that version.

**Comment by alexbozhenko** [comment]
@julieqiu, @jba
Hi!
Seems like last time this logic was modified in:
https://go-review.googlesource.com/c/pkgsite/+/241818

If you agree that this should be fixed, I would like to work on the fix.

**Comment by hyangah** [comment]
I agree that it is a bug. (@jba please confirm so @alexbozhenko can start to work on it)
